controller/volume_controller.go

```diff
@@ -876,6 +881,14 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 		v.Status.Robustness = types.VolumeRobustnessFaulted
 		v.Status.CurrentNodeID = ""
 
+		// Since all replica failed, mark engine controller salvage requested
+		e.Spec.SalvageRequested = true
+		logrus.Infof("All replicas are failed, set engine salvageRequested to %v", e.Spec.SalvageRequested)
+		if e.Status.SalvageExecuted {
+			logrus.Warnf("Engine has executed salvage, execute salvage again")
+			e.Status.SalvageExecuted = false
```

Comment:
Do not update engine status outside of the engine controller.

Reply:
Oh, i see. I will fix it. Thanks.